content/docs/reference/design-docs/signaling.en.md

```diff
@@ -0,0 +1,419 @@
+- provide the **minimum necessary information** to the signaling modules for their operation
+- enable using signaling modules without having to instantiate a complete simulation
+- the design of the signaling module API must allow for signals to be loaded in any order
+
+```yaml
+{
+    # unique identifier for the signaling system
+    "id": "BAL",
+    "version": "1.0",
+    # a list of roles the system assumes
+    "roles": ["MA", "SPEED_LIMITS"],
+    # the schema of the dynamic state of signals of this type
+    "signal_state": [
+        {"kind": "enum", "field_name": "aspect", values: ["VL", "A", "S", "C"]},
+        {"kind": "flag", "field_name": "ralen30"},
+        {"kind": "flag", "field_name": "ralen60"},
+        {"kind": "flag", "field_name": "ralen_rappel"}
+    ],
+    # the schema of the settings signals of this type can read
+    "signal_settings": [
+        {"kind": "flag", "field_name": "Nf", "display_name": "Non-franchissable"},
+        {"kind": "flag", "field_name": "has_ralen30", "default": false, "display_name": "Ralen 30"},
+        {"kind": "flag", "field_name": "has_rappel30", "default": false, "display_name": "Rappel 30"},
+        {"kind": "flag", "field_name": "has_ralen60", "default": false, "display_name": "Ralen 60"},
+        {"kind": "flag", "field_name": "has_rappel60", "default": false, "display_name": "Rappel 60"}
+    ],
+
+    # these are C-like boolean expressions:
+    # true, false, <flag>, <enum> == value, &&, || and ! can be used
+
+    # used to evaluate whether a signal is a block boundary.
+    "block_boundary_when": "true",
+
+    # used to evaluate whether a signal is a route boundary.
+    "route_boundary_when": "Nf",
+
+    # used for naive conflict detection. 
+    "constraining_ma_when": "aspect != VL"
+}
+```
+
+## Assumptions
+
+- Each physical signal can be decomposed into a list of logical signals, all of which are associated with a signaling system.
+- Blocks have a type
+- It is possible to determine, given only the signal, its delimiting properties
+- blocks never cross route boundaries
+- blocks which are not covered by routes do not exist, or can be ignored
+- At any time, trains only use one signaling system capable of transmitting movement authority
+
+# Design of Signaling Systems
+
+Each signaling system has:
+
+- A unique identifier (a string)
+- A set of roles:
+    - Transmission of Movement Authority
+    - Transmission of speed limits
+- Its signal state type, which enables deducing:
+    - The graphical representation of the signal
+    - How a train would react to the signal
+    - If the signal state constrains Movement Authority
+- The signal parameter types, names and description, which enable front-end edition of signal parameters
+- The block and route conditions, which enable evaluating whether a signal delimits blocks or routes, given its parameters.
+
+Note that if a signaling system has a dual role of transmitting MA and speed
```

Comment:
MA has not been explicitely linked to Movement Authority at this point, maybe specify this acronym the first time it's being used.